enhancements/rhcos/split-rhcos-into-layers.md

```diff
@@ -0,0 +1,458 @@
+that can be shared otherwise.
+
+### Implementation Details/Notes/Constraints
+
+As mentioned in the proposal, the work is split into three phases. Here, we go into more
+explicit implementation details for each phase.
+
+##### Phase 1: Using RHEL/CentOS Stream-only bootimages
+
+- In openshift/os, create a new variant of RHCOS which contains solely RHEL packages,
+  i.e. today's RHCOS but without any OCP components.
+  - Note this work has already been done: https://github.com/openshift/os/pull/1445
+- In the RHCOS pipeline, start building this new stream variant. This will push the bootable
+  container to quay.io/openshift-release-dev/ocp-v4.0-art-dev. Bootimages will also be built
+  as usual.
+- Use these bootimages to test changes to the installer boostrapping flow.
+  Notably, bootstrapping currently requires the kubelet and cri-o to be present
+  in order to bring up the temporary control plane. This will be adapted so that
+  the installer first live-applies the node image, as if the node did a system
+  update.
+  - Note this work is already in progress:
+    https://github.com/openshift/installer/pull/8742
+  Other components will also need to be adapted (such as assisted-installer, but
+  likely others too).
+- Work with the teams of the affected components to adapt them.
+- Open an openshift/installer PR which updates the bootimages. This will allow
+  us to run more extensive CI tests on the PR.
+- Raise awareness of this change to the OCP org to invite more testing. 
+- Once all issues have been resolved, merge the PR. If new regressions are
+  encountered, revert the PR and fix them.
+- Once satisfied with the new bootimages, the RHCOS pipeline can stop building
+  the old bootimages containing OpenShift components.
+
+##### Phase 2: Switching `rhel-coreos` in CI
+
+- Set up a new job in openshift/release which builds the layered OpenShift node image on
+  top of the bootable container image from the new stream variant.
+- Open MCO and DTK PRs pointing to this layered image in CI (these are the only two components
+  that explicitly reference `rhel-coreos`) as a way to verify CI is happy.
+    - NOTE: There shouldn't be any major CI failures here. One key thing to understand is that
+      we're changing *how* the node image is built, but not *what* its final contents are from
+      a squashed image perspective. So this is mostly a sanity-check that they are indeed
+      equivalent.
+- Once any issues have been resolved, make the new job push to the `:rhel-coreos` tag of the
+  CI imagestream. This simultaneously requires disabling ART's job which syncs RHCOS to that
+  same tag.
+
+##### Phase 3: Switching `rhel-coreos` in production
+
+- Work with DPTP(?) to update the OCP release page to better handle the new layered approach.
+    - It currently links to diffs in the RHCOS release browser. But now the release browser
+      will only contain the RHEL diff, not the OCP diff in the node layer. We should close this
```

Comment:
This would also break some ART logic. They query the release browser to determine the version of RHEL and OCP package versions in a given version to ensure that the images are up-to-date.